### PR TITLE
refactor: unify trade tokens places switching

### PR DESCRIPTION
**Pull request by shoom3301**

### CODE REVIEW

File: src/modules/trade/hooks/useSwitchTokensPlaces.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { useCallback } from 'react'
+import { FractionUtils } from 'utils/fractionUtils'
+import { ExtendedTradeRawState } from '../types/TradeRawState'
+import { useIsWrapOrUnwrap } from './useIsWrapOrUnwrap'
+import { useTradeNavigate } from './useTradeNavigate'
+import { useWalletInfo } from 'modules/wallet'
+import { useTradeState } from './useTradeState'
+import { useDerivedTradeState } from './useDerivedTradeState'
```
Comment (alfetopito) [inline src/modules/trade/hooks/useSwitchTokensPlaces.ts:8]: Why are the imports not sorted?

File: src/modules/trade/hooks/useSwitchTokensPlaces.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    if (isWrapOrUnwrap) return
+
+    updateState({
+      inputCurrencyId: outputCurrencyId,
+      outputCurrencyId: inputCurrencyId,
+      inputCurrencyAmount: FractionUtils.serializeFractionToJSON(outputCurrencyAmount),
+      outputCurrencyAmount: FractionUtils.serializeFractionToJSON(inputCurrencyAmount),
+      ...stateOverride,
+    })
+    tradeNavigate(chainId, { inputCurrencyId: outputCurrencyId, outputCurrencyId: inputCurrencyId })
```
Comment (alfetopito) [inline src/modules/trade/hooks/useSwitchTokensPlaces.ts:31]: One thing to keep in mind is that if we use this for Swap, there's special handling logic to disable BUY orders for ETH flow in the reducer

Check the usage of this fn https://github.com/cowprotocol/cowswap/blob/628c62596d65e0761ccf70677a55bec9a0a36411/src/legacy/state/swap/reducer.ts#L143

**Review by alfetopito — COMMENTED** [review]
Reviewed only the code.

What are the testing steps?
And PR description :)

File: src/modules/trade/hooks/useSwitchTokensPlaces.ts (hunk 2 above)
Comment (shoom3301) [reply src/modules/trade/hooks/useSwitchTokensPlaces.ts:31]: Thank you! We don't use it for Swap, but we should add TODO at lest for it. Thanks!
@nenadV91 you can include a TODO comment after you merge this

File: src/modules/trade/hooks/useSwitchTokensPlaces.ts (hunk 1 above)
Comment (shoom3301) [reply src/modules/trade/hooks/useSwitchTokensPlaces.ts:8]: It's because I forked from @nenadV91 branch and apparently it doesn't contain new linter rules

Comment (alfetopito) [reply src/modules/trade/hooks/useSwitchTokensPlaces.ts:8]: Oooooh! Makes sense.
Not a problem, it'll be fixed (or force you to fix) when it's merged

**Pull request merged**